src/Components/Components/src/RenderTree/RenderTreeFrame.cs

```diff
@@ -119,6 +133,12 @@ namespace Microsoft.AspNetCore.Components.RenderTree
         /// </summary>
         [FieldOffset(24)] internal readonly ComponentState ComponentState;
 
+        /// <summary>
+        /// If the <see cref="FrameType"/> property equals <see cref="RenderTreeFrameType.Component"/>,
+        /// gets the component's diffing key, or null if none was specified.
+        /// </summary>
+        [FieldOffset(32)] public readonly object ComponentKey;
```

Comment:
I imagine that this value is always set to something "serializable" even though its marked as object given the assumption that we send this to the client.

Reply:
We never serialize or send this object to the client. It doesn't have to be serializable.